.circleci/config.yml

```diff
@@ -0,0 +1,104 @@
+version: 2.1
+
+commands:
+  gem_install_puma:
+    steps:
+      - run: ruby --version
+      - run: gem install puma
+
+  go_test:
+    steps:
+      - checkout
+      - run: go version
+      - run: go mod download
+      - run: go test -v -race -coverprofile=coverage.out -covermode=atomic -timeout=300s ./...
+
+  macos_check_version:
+    steps:
+      - run: sw_vers
+
+  macos_install_go_1_13:
+    steps:
+      - run:
+          name: install go v1.13
+          command: |
+            brew --version
+            cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
+            git fetch
+            git checkout b5c6a2435016eb4e96598c0ada5b1bad61775165 Formula/go.rb
```

Comment:
the version of homebrew that ships in circle's high-sierra image is too old to allow us to install go 1.13. So, manually fetch from the repo, checkout the version we want (that provides 1.13), and install.